tools/bqpipeline/bqpipeline/bqpipeline.py

```diff
@@ -0,0 +1,241 @@
+        self.location = location
+        self.debug = debug
+        if default_project is None and query_project is not None:
+            self.default_project = query_project
+        else:
+            self.default_project = default_project
+        self.json_credentials_path = json_credentials_path
+        self.default_dataset = default_dataset
+        self.bq = None
+        self.j2 = SandboxedEnvironment()
+
+    def get_client(self):
+        if self.bq is None:
+            if self.json_credentials_path is not None:
+                self.bq = bigquery.Client.from_service_account_json(self.json_credentials_path)
+                if self.query_project is not None:
+                    self.bq.project = self.query_project
+                if self.location is not None:
+                    self.bq._location = self.location
+            else:
+                self.bq = bigquery.Client(project=self.query_project, location=self.location)
+        return self.bq
+
+    def resolve_ts(self, dest):
+        table_id = dest
+        if table_id is not None:
+            parts = table_id.split('.')
+            if len(parts) == 2 and self.default_project is not None:
+                table_id = self.default_project + '.' + dest
+            elif len(parts) == 1 and self.default_project is not None and self.default_dataset is not None:
+                table_id = self.default_project + '.' + self.default_dataset + '.' + dest
+        return table_id
+
+    def job_config(self, batch=False, dest=None, create=True, overwrite=True):
+        """
+
+        :param batch: use QueryPriority.BATCH if true
+        :param dest: tablespec of destination table
+        :param create: if False, destination table must already exist
+        :param overwrite: if False, destination table must not exist
+        :return:
+        """
+        dest_tableref = None
+        cd = bigquery.job.CreateDisposition.CREATE_NEVER
+        if create:
+            cd = bigquery.job.CreateDisposition.CREATE_IF_NEEDED
+
+        wd = bigquery.job.WriteDisposition.WRITE_EMPTY
+        if overwrite:
+            wd = bigquery.job.WriteDisposition.WRITE_TRUNCATE
+
+        pr = bigquery.QueryPriority.INTERACTIVE
+        if batch:
+            pr = bigquery.QueryPriority.BATCH
+
+        if dest is not None:
+            dest_tableref = tableref1(dest)
+
+        ds = None
+        if self.default_project is not None and self.default_dataset is not None:
+            ds = bigquery.dataset.DatasetReference(project=self.default_project, dataset_id=self.default_dataset)
+
+        return bigquery.QueryJobConfig(priority=pr,
+                                       default_dataset=ds,
+                                       destination=dest_tableref,
+                                       create_disposition=cd,
+                                       write_disposition=wd)
+
+    def run_query(self, path, batch=False, wait=True, create=True, overwrite=True, timeout=20*60, **kwargs):
```

Comment:
I could default them to None and then load class variables if None but it would add a lot of code, this class is being opinionated on the defaults. I'm considering setting batch to True by default.